Client/Frontend/Settings/HomepageSettings/WallpaperSettings/v2/WallpaperSettingsViewModel.swift

```diff
@@ -0,0 +1,210 @@
+
+    var numberOfSections: Int {
+        return wallpaperCollections.count
+    }
+
+    init(wallpaperManager: WallpaperManagerInterface = WallpaperManager(), tabManager: TabManager) {
+        self.wallpaperManager = wallpaperManager
+        self.tabManager = tabManager
+        setupWallpapers()
+    }
+
+    func numberOfWallpapers(in section: Int) -> Int {
+        return wallpaperCollections[safe: section]?.wallpapers.count ?? 0
+    }
+
+    func sectionHeaderViewModel(for sectionIndex: Int, dismissView: @escaping (() -> Void)) -> WallpaperSettingsHeaderViewModel? {
+        guard let collection = wallpaperCollections[safe: sectionIndex] else { return nil }
+
+        let isClassic = collection.type == .classic
+        let title: String = isClassic ?
+            .Settings.Homepage.Wallpaper.ClassicWallpaper : .Settings.Homepage.Wallpaper.LimitedEditionWallpaper
+        var description: String? = isClassic ? nil : .Settings.Homepage.Wallpaper.IndependentVoicesDescription
+        let buttonTitle: String? = isClassic ? nil : .Settings.Homepage.Wallpaper.LearnMoreButton
+
+        // the first limited edition collection has a different description, any other collection uses the default
+        if sectionIndex > 1 {
+            description = .Settings.Homepage.Wallpaper.LimitedEditionDefaultDescription
+        }
+
+        let buttonAction = { [weak self] in
+            guard let strongSelf = self, let learnMoreUrl = collection.learnMoreUrl else { return }
+
+            dismissView()
+            let tab = strongSelf.tabManager.addTab(URLRequest(url: learnMoreUrl),
+                                                   afterTab: strongSelf.tabManager.selectedTab,
+                                                   isPrivate: false)
+            strongSelf.tabManager.selectTab(tab)
+        }
+
+        let a11yIds = AccessibilityIdentifiers.Settings.Homepage.CustomizeFirefox.Wallpaper.self
+
+        return WallpaperSettingsHeaderViewModel(
+            title: title,
+            titleA11yIdentifier: "\(a11yIds.collectionTitle)_\(sectionIndex)",
+            description: description,
+            descriptionA11yIdentifier: "\(a11yIds.collectionDescription)_\(sectionIndex)",
+            buttonTitle: buttonTitle,
+            buttonA11yIdentifier: "\(a11yIds.collectionButton)_\(sectionIndex)",
+            buttonAction: buttonAction)
+    }
+
+    func updateSectionLayout(for traitCollection: UITraitCollection) {
+        if traitCollection.horizontalSizeClass == .compact {
+            sectionLayout = .compact
+        } else {
+            sectionLayout = .regular
+        }
+        setupWallpapers()
+    }
+
+    func cellViewModel(for indexPath: IndexPath) -> WallpaperCellViewModel? {
+        guard let collection = wallpaperCollections[safe: indexPath.section],
+                let wallpaper = collection.wallpapers[safe: indexPath.row] else {
+            return nil
+        }
```

Comment:
```suggestion
        guard let collection = wallpaperCollections[safe: indexPath.section],
                   let wallpaper = collection.wallpapers[safe: indexPath.row]
        else { return nil }
```
Nit: the else block on a separate line. BUT ALSO: the let is weirdly indented again. lolololol